browser/ui/views/brave_actions/brave_shields_action_view.cc

```diff
@@ -122,6 +122,7 @@ BraveShieldsActionView::GetImageSource() {
   if (is_enabled)
     image_source->SetBadge(std::move(badge));
 
+  SetEnabled(is_enabled); // Disable click event of this label button
```

Comment:
I think `UpdateIconState()` is more better caller for `SetEnabled()` instead of Getter.
Not sure you're still in progress but using `shields_data_controller->GetBraveShieldsEnabled()` for
button enabled status will cause another bug. It'll become not clickable after shields is disabled for normal site(not internal site). I think what we want is only disabling for internal site.

Reply:
yes, i was still working on this. The above solution did introduce another bug and also `SetEnabled` does not have hover state. We're trying to replicate the label button as-is. I think it's better to not render the bubble and keep the button clickable.